src/intent_pilot/utils/config.py

```diff
@@ -83,6 +84,10 @@ def initialize_openai(self, temperature=0.2, max_tokens=4096):
         )
         return client
 
+    def initialize_ollama(self, model="llava", temperature=0):
+        client = ChatOllama(model=model, temperature=temperature)
```

Comment:
A suggestion:
Langchain is criticised in ml community for over engineering and increases overhead in terms of libraries as it tries to do everything.

Ollama supports openai library integration (https://ollama.com/blog/openai-compatibility). what it means is that, we can reuse the function calls like gpt4 by only setting the baseURL to ollama url.
```
client = OpenAI(
    base_url = 'http://localhost:11434/v1',
    api_key='ollama', # required, but unused
)
```

Reply:
I did not know about this actually.

But it does not support vision models yet :/
I tried anyway and yes it throws an error.

What shall we do now? Imho Langchain is okish in this scenario as we do not have production code.

Reply:
true